db/compaction/compaction_job_test.cc

```diff
@@ -265,12 +272,6 @@ class CompactionJobTest : public testing::Test {
     SetIdentityFile(env_, dbname_);
 
     VersionEdit new_db;
-    if (db_options_.write_dbid_to_manifest) {
-      DBImpl* impl = new DBImpl(DBOptions(), dbname_);
-      std::string db_id;
-      impl->GetDbIdentityFromIdentityFile(&db_id);
-      new_db.SetDBId(db_id);
-    }
```

Comment:
This chunk is never tested (otherwise it should have been caught by asan), thus this PR removing it does not reduce test coverage. IMHO, creating a db in a compaction_job_test which does not have the notion of a `db` is itself problematic. We should add separate set of tests for `write_dbid_to_manifest` in a separate PR.